File: java-demo/src/main/java/com/nyt/javademo/SentimentController.java
package com.nyt.javademo;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.util.LinkedMultiValueMap;
import org.springframework.util.MultiValueMap;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.client.RestTemplate;

import java.util.ArrayList;
import java.util.List;

@RestController
@RequestMapping("/api/sentiment")
public class SentimentController {

    private ArticleRepository repository;
    private RestTemplate restTemplate;

    private String sentimentURL;

    public SentimentController(ArticleRepository repository,
                               RestTemplate restTemplate,
                               @Value("${sentiment.url}") String sentimentURL) {
        this.repository = repository;
        this.restTemplate = restTemplate;
        this.sentimentURL = sentimentURL;
    }

    @RequestMapping("/{id}")
    public Sentiment get(@PathVariable Long id) {

        var article = repository.findById(id).get();

        MultiValueMap<String, String> parametersMap = new LinkedMultiValueMap<>();
        parametersMap.add("text", article.getArticleAbstract());

        var sentiment = restTemplate.postForObject(sentimentURL, parametersMap, Sentiment.class);

        sentiment.setTitle(article.getTitle());

        return sentiment;

    }

    @RequestMapping("/")
    public List<Sentiment> all() {
        List<Sentiment> sentiments = new ArrayList<>();
        repository.findAll().forEach((article -> {
            MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
            params.add("text", article.getArticleAbstract());

            Sentiment sentiment = restTemplate.postForObject(sentimentURL, params, Sentiment.class);
            sentiment.setTitle(article.getTitle());
            sentiments.add(sentiment);
        }));

        return sentiments;
    }
}

File: kotlin-demo/src/main/kotlin/com/nyt/kotlindemo/SentimentController.kt
package com.nyt.kotlindemo

import org.springframework.beans.factory.annotation.Value
import org.springframework.util.LinkedMultiValueMap
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.client.RestTemplate

@RestController
@RequestMapping("/api/sentiment")
class SentimentController(val articleRepository: ArticleRepository,
                          val restTemplate: RestTemplate,
                          @Value("\${sentiment.url}") val sentimentURL: String) {

    @RequestMapping("{id}")
    fun get(@PathVariable id: Long): Sentiment {

        val article = articleRepository.findById(id).get()

        val params = LinkedMultiValueMap<String, String>()
        params["text"] = article.articleAbstract

        val sentiment = restTemplate.postForObject(sentimentURL, params, Sentiment::class.java)
        sentiment!!.title = article.title

        return sentiment
    }

    @RequestMapping("/")
    fun all(): List<Sentiment> {
        var sentiments = mutableListOf<Sentiment>()

        articleRepository.findAll().forEach {
            val params = LinkedMultiValueMap<String, String>()
            params["text"] = it.articleAbstract

            var sentiment = restTemplate.postForObject(sentimentURL, params, Sentiment::class.java)
            sentiment!!.title = it.title

            sentiments.add(sentiment)
        }

        return sentiments
    }
}